app/assets/javascripts/discourse/controllers/composer.js.es6

```diff
@@ -109,10 +110,29 @@ export default Ember.Controller.extend({
             !creatingPrivateMessage;
   },
 
-  @computed('model.action')
-  canWhisper(action) {
+  @computed('model.whisper', 'model.unlistTopic')
+  whisperOrUnlistTopicText(whisper, unlistTopic) {
+    if (whisper) {
+      return I18n.t("composer.whisper");
+    } else if (unlistTopic) {
+      return I18n.t("composer.unlist");
+    }
+  },
+
+  @computed
+  isStaffUser() {
     const currentUser = this.currentUser;
-    return currentUser && currentUser.get('staff') && this.siteSettings.enable_whispers && action === Composer.REPLY;
+    return currentUser && currentUser.get('staff');
+  },
+
+  @computed('model.creatingTopic', 'isStaffUser')
+  canUnlistTopic(creatingTopic, isStaffUser) {
```

Comment:
I believe this can also be written as `canUnlistTopic: Ember.computed.and("model.creatingTopic", "isStaffUser")` ;)